saml2/saml2-service-provider/src/main/java/org/springframework/security/saml2/provider/service/authentication/logout/Saml2LogoutRequestAuthenticator.java

```diff
@@ -0,0 +1,38 @@
+/*
+ * Copyright 2002-2021 the original author or authors.
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *      https://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+
+package org.springframework.security.saml2.provider.service.authentication.logout;
+
+/**
+ * Authenticates SAML 2.0 Logout Requests
+ *
+ * @author Josh Cummings
+ * @since 5.6
+ */
+public interface Saml2LogoutRequestAuthenticator {
```

Comment:
I'm still not a fan of `Authenticator`. To me this is a validator. I recall you didn't like a validator because you said it verifies and validates, but in this case verify could be replaced with validate signature.